lib/ansible/parsing/vault/__init__.py

```diff
@@ -123,16 +135,25 @@ def is_encrypted(b_data):
 def is_encrypted_file(file_obj):
     """Test if the contents of a file obj are a vault encrypted data blob.
 
-    The data read from the file_obj is expected to be bytestrings (py2 'str' or
-    python3 'bytes'). This more or less expects 'utf-8' encoding.
-
     :arg file_obj: A file object that will be read from.
     :returns: True if the file is a vault file. Otherwise, False.
     """
     # read the header and reset the file stream to where it started
     current_position = file_obj.tell()
-    b_header_part = file_obj.read(len(b_HEADER))
-    file_obj.seek(current_position)
+    try:
+        file_obj.seek(0)
```

Comment:
Hmm... That's an interesting point.  If we don't seek(0) then we depend on the calling code to do the right thing when setting up the filehandle.  OTOH, it allows calling code to do things like "read this vault data from the middle of this email message".  What if we add a new parameter?
``` python
def  is_encrypted_file(file_obj, file_pos=0):
    [...]
    file_obj.seek(file_pos)
```